WISE-Community/WISE
# Added support to format TypeScript files with Prettier

Also added npm script to batch format files in the future.

This will make sure that our TypeScript files are formatted consistently on save and paste.

Note about trying to implement [Google's indent continuation lines at least +4 spaces guideline](https://google.github.io/styleguide/jsguide.html#formatting-indent) : I could not find an easy way to add 4 spaces at the beginning of a continuation line. The default behavior is 2 spaces. I also found this issue on Prettier: https://github.com/prettier/prettier/issues/5865. It appears that Prettier by default puts the long line into separate lines. This is for function parameters though, e.g.

```
  getMultipleAttemptCRaterFeedbackTextByScore( component: any, previousScore: any, currentScore: any) {
```
becomes
```
  getMultipleAttemptCRaterFeedbackTextByScore(
    component: any,
    previousScore: any,
    currentScore: any
  ) {
```

Perhaps we can go with the default Prettier formatting for now. What do you think?

How to test:
1. download Prettier plugin for VSCode, if you haven't already.
2. make sure that VSCode's default formatter is set to "esbenp.prettier-vscode", and not set to null
<img width="567" alt="Screen Shot 2021-01-06 at 10 32 44 AM" src="https://user-images.githubusercontent.com/119416/103806924-8d924300-500a-11eb-8a18-343968cf13f5.png">
3. make sure that formatting on save and on paste are enabled in VSCode. Follow this guide: https://khalilstemmler.com/blogs/tooling/prettier/#Formatting-using-VSCode-on-save-recommended. 
4. test that the file formats when you save, or paste in a block of code into the file.


Closes #2861
